### PR TITLE
Add Alembic migration to create initial v2 database

### PR DESCRIPTION
**Pull request by rod-glover**
This is in support of the larger project of bringing `modelmeta` entirely under Alembic migration management, and the more specific goal of enabling a new instance of a `modelmeta` database to be created using Alembic.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -9,30 +9,195 @@ modelmeta
...
 
---------------
-How to Install
---------------
+``modelmeta`` uses `SQLAlchemy <http://www.sqlalchemy.org>`_ to provide the ORM layer, and
+`Alembic <http://alembic.zzzcomputing.com/en/latest/>`_ to manage database creation and migration (see section
+below).
 
-One can install `modelmeta` using the standard methods of any other Python package.
+The intent of the database itself is to separate the small, inexpensive, structured metadata and attribute information
```
Comment (jameshiebert) [inline README.rst]: Good addition.

File: README.rst [hunk 2]
```diff
@@ -9,30 +9,195 @@ modelmeta
...
+Installation
+============
+
+Install ``modelmeta`` using the standard methods for any other Python package.
+
+1. Clone the repository and run the setup script::
 
     $ git clone https://github.com/pacificclimate/modelmeta
     $ cd modelmeta
     $ python setup.py install
```
Comment (jameshiebert) [inline README.rst]: May want to add the pip instructions (e.g. ``pip install -e .`` or ``pip install .``), while we're working on this section.

File: README.rst [hunk 3]
```diff
@@ -9,30 +9,195 @@ modelmeta
...
+
+History
+-------
+
+The existing instance of a ``modelmeta`` database (``monsoon/pcic_meta``) was created prior to the adoption of
+Alembic, and therefore the timeline for Alembic database migrations is slightly confusing.
+
+Timeline:
+
+- *the distant past*: ``pcic_meta`` is created by mysterious primeval processes.
```
Comment (jameshiebert) [inline README.rst]: Haha. That or typing SQL into a terminal...

File: README.rst [hunk 4]
```diff
@@ -9,30 +9,195 @@ modelmeta
...
+      database schema from an empty database.
+
+    - The add-seasonal migration is modified to logically follow the initial-create migration.
+
+Creating a new ``modelmeta`` database
+-------------------------------------
+
+There are in fact two tools in this package to create a new ``modelmeta`` database.
+
+- (DEPRECATED) Use the script ``scripts/mkblankdb.py``.
```
Comment (jameshiebert) [inline README.rst]: Should we just remove it if it's deprecated?

File: README.rst [hunk 5]
```diff
@@ -9,30 +9,195 @@ modelmeta
...
+
+        - the username is the same as the schema, e.g., ``ce_meta``,
+          with the default search path (``"$user",public``), OR
+        - it has a default search path of the form ``<name>,public``, e.g., ``ce_meta,public``
+
+    #. Create a new database with the chosen name, e.g., ``ce_meta``.
+
+    #. Within that database, create a new schema with the chosen name, e.g., ``ce_meta``.
+
+    #. Somehow install PostGIS in there. How??
```
Comment (jameshiebert) [inline README.rst]: Pretty much comes down to issuing ``CREATE EXTENSION POSTGIS`` at this point, but you could probably link to the PostGIS install docs.

File: README.rst [hunk 6]
```diff
@@ -9,30 +9,195 @@ modelmeta
...
+    alembic -x db=prod_ce_meta upgrade head
+
+#. Have a beer.
+
+For a SQLite database
++++++++++++++++++++++
+
+A SQLite database is very simple to set up, but is normally used only for testing.
+
+#. Add a DSN for your new database to ``alembic.ini``. This database need not exist yet. For example::
```
Comment (jameshiebert) [inline README.rst]: The file need not exist, but presumably the path *does*?

File: alembic.ini [hunk 7]
```diff
@@ -22,7 +20,7 @@ script_location = alembic
 # revision_environment = false
 
 # set to 'true' to allow .pyc and .pyo files without
-# a source .py file to be detected as revisions in the
+# a source .py file to be detected as revisions in thqe
```
Comment (jameshiebert) [inline alembic.ini]: Typo.

File: modelmeta/v2.py [hunk 8]
```diff
@@ -56,6 +63,10 @@ class DataFile(Base):
     def __str__(self):
         return '<DataFile %s>' % self.filename
 
+UniqueConstraint(DataFile.unique_id, name='data_files_unique_id_key')
```
Comment (jameshiebert) [inline modelmeta/v2.py]: Thank you :)

File: modelmeta/v2.py [hunk 9]
```diff
@@ -140,6 +155,51 @@ class Grid(Base):
     yc_origin = Column(Float, nullable=False)
     yc_units = Column(String(length=64), nullable=False)
 
+    # Brace yourself.
```
Comment (jameshiebert) [inline modelmeta/v2.py]: Well... it works I guess.

File: README.rst (hunk 4 above)
Comment (rod-glover) [inline README.rst]: I'm good with that. I didn't want to delete it until I checked with you that it didn't retain some use. I'll take that as a 'no'.

**Pull request merged**